### PR TITLE
etcdmanager / Etcd Deployment

### PR DESCRIPTION
**Pull request by eberlep**

### CODE REVIEW

File: external/svc-etcd.yaml [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+    - update
+  - apiGroups:
+    - ""
+    resources:
+    - nodes
+    verbs:
+    - get
+    - list
+    - watch
+  # - apiGroups:
```
Comment (majst01) [inline external/svc-etcd.yaml:73]: Delete before release

File: external/svc-etcd.yaml [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+    - create
+    - delete
+    - get
+    - patch
+    - update
+  - apiGroups:
+    - apps
+    resources:
+    - statefulsets
+    # - deployments
```
Comment (majst01) [inline external/svc-etcd.yaml:122]: delete

File: external/svc-etcd.yaml [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+- kind: Role
+  apiVersion: rbac.authorization.k8s.io/v1
+  metadata:
+    labels:
+      postgres.database.fits.cloud/partition-id: sample
+      postgres.database.fits.cloud/managed-by: postgreslet
+      postgres.database.fits.cloud/component: etcd
+    name: patroni-etcd
+    namespace: postgreslet-system
+  rules:
```
Comment (majst01) [inline external/svc-etcd.yaml:25]: Review of required resources necessary

Comment (eberlep) [reply external/svc-etcd.yaml:25]: Turns out, almost none of it is neccessary :-)

File: main.go [hunk 4]
```diff
@@ -207,6 +220,20 @@ func main() {
 	viper.SetDefault(ControlPlaneNamespaceFlg, "metal-extension-postgres")
 	controlPlaneNamespace = viper.GetString(ControlPlaneNamespaceFlg)
 
+	viper.SetDefault(deployEtcdFlg, false)
+	deployEtcd = viper.GetBool(deployEtcdFlg)
+
+	etcdImage = viper.GetString(etcdImageFlg)
+	etcdBackupSidecarImage = viper.GetString(etcdBackupSidecarImageFlg)
+	viper.SetDefault(etcdBackupSecretNameFlg, "etcd-backup-restore-s3-config") // TODO remove
```
Comment (majst01) [inline main.go:228]: rename accordingly

File: main.go [hunk 5]
```diff
@@ -267,6 +300,31 @@ func main() {
...
+	}
+	etcdMgr, err := etcdmanager.New(svcClusterConf, "external/svc-etcd.yaml", scheme, ctrl.Log.WithName("EtcdManager"), etcdMgrOpts)
+	if err != nil {
+		setupLog.Error(err, "unable to create `EtcdManager`")
+		os.Exit(1)
+	}
+	if deployEtcd {
+		if err = etcdMgr.InstallOrUpdateEtcd(); err != nil {
+			setupLog.Error(err, "unable to deploy etcd")
+			os.Exit(1) // TODO should we really exit here?
```
Comment (majst01) [inline main.go:320]: remove TODO i prefer to fail early.

File: pkg/etcdmanager/etcdmanager.go [hunk 6]
```diff
@@ -0,0 +1,560 @@
...
+	SecretKeyRefName       string
+	PSPName                string
+	PostgresletFullname    string
+}
+
+// OperatorManager manages the operator
+type EtcdManager struct {
+	client.Client
+	runtime.Decoder
+	list *corev1.List
```
Comment (majst01) [inline pkg/etcdmanager/etcdmanager.go:55]: this is actually the deserialized etcd manfest, therefore the field name is wrong

**Review by majst01 — CHANGES_REQUESTED** [review]

**Pull request merged**